backend/tests/unit/dal/Services/Admin/AgencyServiceTest.cs

```diff
@@ -0,0 +1,330 @@
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.SystemAdmin);
+            using var init = helper.InitializeDatabase(user);
+            var agency1 = EntityHelper.CreateAgency(100, "AG1", "Agency 1");
+            init.AddAndSaveChanges(agency1);
+            var expectedCount = 1;
+
+            var service = helper.CreateService<AgencyService>(user);
+
+            // Act
+            var result = service.Get(1, 1);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.IsAssignableFrom<Paged<Entity.Agency>>(result);
+            Assert.Equal(expectedCount, result.Items.Count());
+        }
+
+        [Fact]
+        public void Get_Agencies_NotAuthorized()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission();
+
+            var service = helper.CreateService<AgencyService>(user);
+
+            // Act
+            // Assert
+            Assert.Throws<NotAuthorizedException>(() =>
+                service.Get(1, 1));
+        }
+
+        [Theory]
+        [MemberData(nameof(AgencyFilterData))]
+        public void Get_Agencies(AgencyFilter filter, int expectedCount)
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.SystemAdmin);
+
+            using var init = helper.InitializeDatabase(user);
+            var agency1 = EntityHelper.CreateAgency(100, "TST", "Test");
+            agency1.ParentId = 0;
+            agency1.IsDisabled = false;
+            var agency2 = EntityHelper.CreateAgency(101);
+            agency2.ParentId = 0;
+            var agency3 = EntityHelper.CreateAgency(102, "TST3", "Agency 3");
+            agency3.IsDisabled = true;
+            var agency4 = EntityHelper.CreateAgency(200, "TST4", "Agency 4");
+
+            init.AddAndSaveChanges(agency1, agency2, agency3, agency4);
+
+            var service = helper.CreateService<AgencyService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            var result = service.Get(filter);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.IsAssignableFrom<Paged<Entity.Agency>>(result);
+            Assert.Equal(expectedCount, result.Items.Count());
+        }
+
+        [Fact]
+        public void GetAll_Agencies()
+        {
+            // Arrange
+            var helper = new TestHelper();
+            var user = PrincipalHelper.CreateForPermission(Permissions.SystemAdmin);
+
+            using var init = helper.InitializeDatabase(user);
+
+            var service = helper.CreateService<AgencyService>(user);
+            var context = helper.GetService<PimsContext>();
+
+            // Act
+            var result = service.GetAll();
+
+            // Assert
+            Assert.Equal(18, result.Count());
```

Comment:
Might be wise to generate the count dynamically so this doesn't fail when someone updates the InitializeDatabase